signature-specification.md

```diff
@@ -148,6 +170,12 @@ The client implementation can use the aforementioned `mediaType` to parse the si
 [COSE_Sign1_Tagged](./signature-envelope-cose.md) signature envelope versioning is achieved by the `cty` field in ProtectedHeaders.  
 [JWS JSON serialization](./signature-envelope-jwt.md) signature envelope versioning is achieved by the `cty` field in ProtectedHeaders.
 
+## Appendix
+
+### Signing time
+
+The signing time denotes the time at which the signature was generated. A X509 certificate has a defined lifetime during which it can be used to generate signatures. Signatures generated after the certificate expires are considered invalid. A trusted timestamp allows a verifier to determine if the signature was generated when the certificate was valid. It also allows a verifier to determine if a signature be treated as valid when a certificate is revoked, if the certificate was revoked after the signature was generated. In the absence of a trusted timestamp, signatures are considered invalid after certificate expires, and all signatures are considered revoked when a certificate is revoked.
```

Comment:
> It also allows a verifier to determine if a signature be treated as valid when a certificate is revoked, if the certificate was revoked after the signature was generated.

If the certificate is revoked, the signature should also be revoked even if it is generated before the revocation.

Here's a typical scenario: Eve hacks Alice's computer and get Alice's private key. Eve uses Alice's key to generate a signature. After a few hours, Alice detects the intrusion and revoked her certificate. Apparently, Alice wants the signatures generated by Eve to be revoked, even with the cost of revoking Alice's genuine signatures.

Reply:
This is usually the signer's choice. Alice may request the CA to revoke the certificate outright (all signatures are considered invalid), or from a particular date (e.g. X.509 cert [InvalidityDate](https://datatracker.ietf.org/doc/html/rfc5280#section-5.3.2)). This allows the signer to dictate which behavior they want, `InvalidityDate` may not be widely supported in verification environments, I've seen few examples which honor it.